command/server.go

```diff
@@ -1685,6 +1685,11 @@ func (c *ServerCommand) Run(args []string) int {
 				}
 			}
 
+			// notify ServiceRegistration that a configuration reload has occurred
+			if sr := coreConfig.GetServiceRegistration(); sr != nil {
+				sr.NotifyConfigurationReload(config.ServiceRegistration.Config)
```

Comment:
It appears that ServiceRegistration _can_ be nil, which means that this line can lead to a panic in that case.

Reply:
If it's not too much more work handling the removal of the service_registration configuration, I think it makes sense, because it seems to me that it would be the next logical leap once modifications can be reloaded.

Reply:
Updating the code in the latest commit to handle this case which works when testing locally with a manual build when consul ACL is disabled; however, I can't get the test to pass because it can't seem to deregister the consul service from the catalog.

I am able to reproduce this same behavior on main using

```hcl
# config.hcl
acl {
  enabled = true
  default_policy = "deny"

  tokens {
    initial_management = "e95538d0-5cb1-4245-a217-338ed6f7dae8"
  }
}
```

```
consul agent -dev -config-file config.hcl
```

```hcl
# service_registration.hcl
service_registration "consul" {
  token = "e95538d0-5cb1-4245-a217-338ed6f7dae8"
}
```

```
make dev-ui && VAULT_LOG_LEVEL=debug ./bin/vault server -dev -config service_registration.hcl
```